src/simulations/modeling/reframed_helpers.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2018 Novo Nordisk Foundation Center for Biosustainability, DTU.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+def create_metabolite_id2name_mapping(external_metabolite_ids, community):
+    """
+        Return a dictionary mapping metabolite IDs to names.
+
+        Parameters
+        ----------
+        external_metabolite_ids: list
+            List of external metabolite IDs generated by reFramed's
+            solution.community.merged_model.get_external_metabolites()
+        community: reframed.community.model.Community
+            Community model object holding respective individual models in
+            addition to merged model and convenience methods.
+
+        Returns
+        -------
+        dict
+            A mapping of all external metabolite IDs to the corresponding
+            metabolite names.
+        """
+    ids_to_names = {}
+    for metabolite_id in external_metabolite_ids:
+        for org in community.organisms:
+            try:
+                ids_to_names[metabolite_id] = (
+                    community.organisms[org].metabolites[metabolite_id].name
+                )
+                break
+            except KeyError:
+                continue
+    return ids_to_names
+
+
+def generate_transactions(metabolite_id2name_dict, exchanges, abstol=1e-6):
+    """
+        Return a list of tuples with each tuple indicating metabolite
+        transactions as (from_org_id, to_org_id, metabolite_id,
+        metabolite_name, flux).
+
+        Parameters
+        ----------
+        metabolite_id2name_dict: dict
+            A mapping of all external metabolite IDs to the corresponding
+            metabolite names.
+        exchanges: dict
+            Mapping between a tuple of (org_id, metabolite_id) and the
+            corresponding flux.
+            A negative flux means uptake and a positive flux means secretion of
+            the corresponding metabolite by the corresponding organism.
+
+        Returns
+        -------
+        list
+            A list of tuples with each tuple indicating metabolite
+            transactions as (from_org_id, to_org_id, metabolite_id,
+            metabolite_name, flux).
```

Comment:
How about instead of tuples returning namedtuples, dicts or similar? It would let you access the properties by the names you've listed here, instead of indices. It can be a bit hard to read what's going on - looking at your tests for example, they're well implemented and do what they're supposed to, but reading them requires a bit of mental mapping.

Reply:
True, good point, since it's just an internal thing I could return named tuples.